state/protocol/inmem/convert.go

```diff
@@ -74,11 +76,24 @@ func FromSnapshot(from protocol.Snapshot) (*Snapshot, error) {
 	}
 	snap.Params = params.enc
 
+	protocolEpochState, err := from.EpochProtocolState()
+	if err != nil {
+		return nil, fmt.Errorf("could not get protocol epoch state: %w", err)
+	}
+	snap.EpochProtocolState = protocolEpochState.Entry().ProtocolStateEntry
```

Comment:
⚠️ ❓ 

I am not sure this is correct, because we are loosing the updates related to node ejections I think (see [my comment above](url) for more details)

Reply:
Shouldn't be the case, since the `Entry` is from a concrete block which is associated with current snapshot.

Reply:
see [discussion above](https://github.com/onflow/flow-go/pull/5650#discussion_r1562513689)